core/src/test/java/nu/yona/server/crypto/seckey/CryptoSessionTest.java

```diff
@@ -83,46 +80,49 @@ public void decrypt_validPassword_returnsDecryptedData(String plaintext)
 		assertThat(result, equalTo(plaintext));
 	}
 
-	@Test(expected = CryptoException.class)
 	public void decrypt_invalidCryptoVariantNumber_throws()
 	{
 		byte[] initializationVector = new byte[INITIALIZATION_VECTOR_LENGTH];
 		byte[] ciphertext = Base64.getDecoder().decode(encrypt(PASSWORD1, PLAINTEXT1, initializationVector));
 		ciphertext[0] = 13; // Unsupported crypto variant number
 
-		decrypt(PASSWORD1, Base64.getEncoder().encodeToString(ciphertext), initializationVector);
+		assertThrows(CryptoException.class,
+				() -> decrypt(PASSWORD1, Base64.getEncoder().encodeToString(ciphertext), initializationVector));
 	}
 
-	@Test(expected = CryptoException.class)
 	public void decrypt_invalidPassword_throws()
 	{
 		byte[] initializationVector = new byte[INITIALIZATION_VECTOR_LENGTH];
 		String ciphertext = encrypt(PASSWORD1, PLAINTEXT1, initializationVector);
 
-		String plaintext = decrypt(PASSWORD2, ciphertext, initializationVector);
+		assertThrows(CryptoException.class, () -> {
+			String plaintext = decrypt(PASSWORD2, ciphertext, initializationVector);
 
-		// In rare cases, decryption with a wrong password doesn't throw but delivers rubbish.
-		// In such rare cases, compare the string and explicitly throw that exception.
-		assertThat(plaintext, not(equalTo(PLAINTEXT1)));
-		throw CryptoException.decryptingData();
+			// In rare cases, decryption with a wrong password doesn't throw but delivers rubbish.
+			// In such rare cases, compare the string and explicitly throw that exception.
+			assertThat(plaintext, not(equalTo(PLAINTEXT1)));
+			throw CryptoException.decryptingData();
+		});
 	}
 
 	@Test
 	public void start_passwordCheckerSaysOk_doesNotThrow()
 	{
 		try (CryptoSession cryptoSession = CryptoSession.start(Optional.of(PASSWORD1), CryptoSessionTest::passwordIsOk))
 		{
-			assertTrue(true);
+			assertThat("Password checker returned OK", true);
```

Comment:
This is a really strange use of `assertThat`. I think it should be `org.junit.jupiter.api.Assertions.assertDoesNotThrow;`

Reply:
You're right. I'll improve it.

Reply:
Why shouldn't we use `assertDoesNotThrow` here?

Reply:
Tests always fail when they throw an exception. ``assertDoesNotThrow`` is meant for special cases only. That's why I opted for explicitly asserting we passed through the try-block. I feel that expresses more explicitly what we want to accomplish.

Reply:
Actually the thing we want to assert is that some code does not throw an exception, that is also what the method name says. `assertDoesNotThrow` is exactly meant for this purpose.

Reply:
``assertDoesNotThrow`` is designed to put an assert around a specific piece of a test, not an entire test, see [the documentation](https://junit.org/junit5/docs/5.3.0/api/org/junit/jupiter/api/Assertions.html#assertDoesNotThrow(org.junit.jupiter.api.function.Executable)). I'll remove the assert and add a comment.

Reply:
My opinion would be that every test should have an assert and as the test ends with `_doesNotThrow` it is logical to use `assertDoesNotThrow`. Something like:
```
CryptoSession cryptoSession;

assertDoesNotThrow(() -> { cryptoSession = CryptoSession.start(Optional.of(PASSWORD1), CryptoSessionTest::passwordIsOk) })

cryptoSession.close();
```

But as the code works I'll close this conversation.